service-print/src/main/java/org/oskari/print/PDF.java

```diff
@@ -735,22 +735,22 @@ private static void draw(PDPageContentStream stream, Geometry g, PDPrintStyle st
         }
     }
 
-    private static void drawLabel(PDPageContentStream stream, Geometry g, String label) throws IOException {
-        Coordinate c;
+    private static void drawLabel(PDPageContentStream stream, PDPrintStyle.LabelAlign align, Geometry g, String label) throws IOException {
+        Coordinate c = null;
         if (g instanceof MultiPoint || g instanceof MultiPolygon ) {
             for (int i = 0 ; i < g.getNumGeometries(); i++){
                 c = g.getGeometryN(i).getCentroid().getCoordinate();
```

Comment:
This actually makes no sense - this is picking the centroid of the last point or polygon in the multipoint/-polygon.

Reply:
If that's the desired functionality then do something like `if (g.isEmpty()) { return; } g = g.getGeometryN(g.getNumGeometries() - 1); // Now g is Point, LineString or Polygon (unless it was GeometryCollection, should check that first)`

Reply:
This was mistake. Looked that there was copy-paste code. Rolled back.